boxes.lisp

```diff
@@ -589,3 +589,75 @@ N.B., Successive calls may return the same object."
   (floor (height box) 2))
 
 
+;;;;;;;;;;;;;;;;;;;;;;;;;;;; ARRAY BOXES ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(defclass array-box (box)
+  ((row-padding :initarg :row-padding
+                :type (integer 0)
+                :reader array-box-row-padding)
+   (row-alignments :initarg :row-alignments
+                   :reader array-box-row-alignments
+                   :type vector ;; vector of VERTICAL-ALIGNMENT
+                   :documentation "Contents can be aligned :TOP, :BASELINE (default), :MIDDLE, or :BOTTOM.")
+   (column-padding :initarg :column-padding
+                   :type (integer 0)
+                   :reader array-box-column-padding
+                   :documentation "The spacing between each column.")
+   (column-alignments :initarg :column-alignments
+                      :reader array-box-column-alignments
+                      :type vector ;; vector of HORIZONTAL-ALIGNMENT
+                      :documentation "Contents can be aligned :LEFT (default), :MIDDLE, or :RIGHT.")
+   (contents :initarg :contents
+             :accessor array-box-contents)
+   (cell-dimensions :initarg :cell-dimensions
+                    :reader array-box-cell-dimensions
+                    ;; array of (HEIGHT . WIDTH) conses.
+                    ))
+  (:default-initargs :row-padding 0
+                     :column-padding 1)
+  (:documentation "A vertical concatenation of boxes."))
+
+(defun make-array-box (rows cols)
```

Comment:
The other boxes define a constructor named `box-<type>` that takes a list of boxes as the box's contents. Why is it different here?

Reply:
this was just for testing